### PR TITLE
Admin Unit Tests

### PR DESCRIPTION
**Pull request by pspani**
Requirements
adding unit tests to admin api

Description of the Change
added unit tests to the jazz_admin service module with 90% coverage

Benefits
the module was missing unit tests.

Possible Drawbacks
Some minor changes were made to the jazz_admin module for enabling the unit tests. This would need to be tested after deploying.

Applicable Issues

### CODE REVIEW

File: core/jazz_admin/test/test.js [hunk 1]
```diff
@@ -1,11 +1,171 @@
...
+const request = require('request');
+
 const index = require('../index');
+const awsContext = require('aws-lambda-mock-context');
+const configObj = require('../components/config.js');
+
+describe('jazz_admin', function () {
+
+  beforeEach(function () {
+    spy = sinon.spy();
```
Comment (svidhani) [inline core/jazz_admin/test/test.js]: why do you need this? r u using it anywhere?

File: core/jazz_admin/test/test.js [hunk 2]
```diff
@@ -1,11 +1,171 @@
...
+        expect(error).to.include('{"errorType":"Unauthorized","message":"This user is not authorized to access this service.');
+      });
+    });
+  });
+
+  describe('getInstallerVarsJSON', () => {
+
+    it("should indicate error while making request to gitlab repo", () => {
+      config.SCM_TYPE = "gitlab";
+      var responseObj = {
```
Comment (svidhani) [inline core/jazz_admin/test/test.js]: try to use const/let, etc.

File: core/jazz_admin/test/test.js [hunk 3]
```diff
@@ -1,11 +1,171 @@
...
+        }
+      };
+      reqStub = sinon.stub(request, "Request").callsFake((obj) => {
+        return obj.callback(null, responseObj, responseObj.body)
+      });
+      index.getInstallerVarsJSON(config)
+        .catch(error => {
+          expect(error).to.include('Unauthorized');
+        });
+      sinon.assert.calledOnce(reqStub);
```
Comment (svidhani) [inline core/jazz_admin/test/test.js]: you should put these statements(sinon.assert and restore inside the callback and not outside.

`.catch(error => {
          expect(error).to.include('Unauthorized');
          sinon.assert.calledOnce(reqStub);
          reqStub.restore();
  })`


fix it everywhere

File: core/jazz_admin/test/test.js (hunk 1 above)
Comment (pspani) [reply core/jazz_admin/test/test.js]: removed. I had initially thought I needed it but forgot to remove it when I didn't need it.

**Pull request merged**